Rakefile

```diff
@@ -109,7 +109,7 @@ end
 def run_tests(workspace, scheme, destination)
   settings = "GCC_PREPROCESSOR_DEFINITIONS='${inherited} ENCRYPT_DATABASE=1'" unless !ENV["encrypted"]
   logName = (ENV["encrypted"] ? "Encrypted" : "") + "#{scheme}.log"
-  return system("xcodebuild -verbose -workspace #{workspace} -scheme '#{scheme}' -destination '#{destination}' #{settings} test | tee #{logName} | xcpretty -r junit; exit ${PIPESTATUS[0]}")
+  return system("xcodebuild -verbose -workspace #{workspace} -scheme '#{scheme}' -destination '#{destination}' #{settings} test | tee -a #{logName} | xcpretty -r junit; exit ${PIPESTATUS[0]}")
```

Comment:
I'd prefer to see an independent log name rather than this append. The combinations of which branch runs on which node will potentially mean different orderings or an overwrite when the artifact is attached to Jenkins regardless of whether we appended on the box or not, and that's not accounting for the case where we may not start from a clean checkout.
Whilst this was fine for getting what you needed I think it is likely to come back and bite us later if we don't do something more robust (I'm willing for it to happen under another PR as it is somewhat unrelated having been introduced by the IAM changes rather than this work, and only being exposed by the need to use the logs here).

Reply:
I think we can find a better way of naming these logs, maybe jenkins should pass in a unique name fore this matrix combination.

Reply:
As discussed I'll back this out and cover it in another PR.